### ISSUE TITLE
PostSharp No Longer Works with Blazor Web Assembly in .NET 8

### ISSUE THREAD

**Issue by PaulSorauer-IPG** [postsharp/PostSharp.IL#6]
Hi there

I've created an issue in the Samples repo as I feel the issue is better demonstrated there.

However, it seems that PostSharp no longer works in Blazor after the Web Assembly app can been upgraded to .NET 8.

Here's the issue I posted in the Samples repo:

[PostSharp No Longer Works with Blazor Web Assembly in .NET 8](https://github.com/postsharp/PostSharp.Samples/issues/21#issue-2150391473)

**Comment by addabis** [comment]
I can confirm that there is some problem with MSBuild integration that was not captured by our automated tests.

We will be working on a fix.

(internal id of this ticket is #34771)

**Comment by addabis** [comment]
Note that I have been able to update our sample repo to work on .NET 8.0. If any problem persists in your application, it means that it is a different problem than we've encountered in samples and we should continue the diagnosis.

**Comment by PaulSorauer-IPG** [comment]
Hi @addabis 

I've carried on the conversation in the other thread, but I am indeed still having issues although I do feel it may still be the same underlying problem.

**Comment by PaulSorauer-IPG** [comment]
Great! It works! 😍

Thanks again @addabis for such a speedy response.

As per the other thread, I did perform a clean at the repo / git level instead of just Visual Studio, I do note that I never had to switch .NET versions in my repo like we did for your Samples repo - I was already at .NET 8.0.x and simply added PostSharp via NuGet to the existing solution.

I think the issue was that I was lacking the `PrivateAssets="none" ` in the PackageReference, but it was also really helpful to know how to make the build break by creating something that PostSharp couldn't build as a way of testing that PostSharp is indeed intercepting the build process.

Thanks for your help on this! I really appreciate it.

Cheers, Paul
